### PR TITLE
[CYS] Update the intro page

### LINKED ISSUE #46837: CYS > Remove the restriction to TT4 and allow users to proceed to the pattern assembler if they have a block theme enabled
**Issue Description:**

As part of the updated Intro screen experience, we need to ensure that whenever a user has a block theme enabled, they are directed to the Assembler. In other words, we need to lift the restriction to the TT4 theme we currently have in place on both [`customize-store/design-without-ai/services.ts`](https://github.com/woocommerce/woocommerce/blob/f9c0d406ae339de148832e06a56a2fc540ce7f93/plugins/woocommerce-admin/client/customize-store/design-without-ai/services.ts#L131) and [`customize-store/design-with-ai/services.ts`](https://github.com/woocommerce/woocommerce/blob/f9c0d406ae339de148832e06a56a2fc540ce7f93/plugins/woocommerce-admin/client/customize-store/design-with-ai/services.ts#L435). 

Additionally, whenever accessing the CYS intro page with a **customized block theme enabled**, the design should be updated to:

Screenshot 2024-04-12 at 11 12 05

Figma: FlPl7fHaGv9mpu8Nfl0joH-fi-2285_15094

As a quick reminder, we have currently established that a theme is considered "customized" only if the user customized their templates or global styles in the editor (See https://github.com/woocommerce/woocommerce/pull/45792 for details).

⚠️ **PS: whenever the work for this issue is complete & ready to merge, if we still didn't come up with an action plan regarding extensibility, let's put it on hold until we do to.**

## Scope

This issue can be marked as complete as soon as:

- The CYS flow is accessible to **any block theme and not just TT4**
- If the currently active block theme was already customized, the intro page should be updated to the design linked on this issue (you can refer to the screenshot as an example).
- Whenever users click on "Customize your theme" with a customized block theme enabled, they should be led to the pattern assembler. 
- Whenever users click on "Customize your theme" with a classic theme enabled, they should be led to the Customizer.
- Whenever users click on "Browse themes" they should be led to WooCommerce → Extensions → Themes.
- Whenever users click on "Start designing" they should be led to the Assembler.
- Bonus points: currently we are setting the default theme both on `design-with-ai/services.ts` and `design-without-ai/services.ts`, let's explore consolidating the updated version to a single source.

### PR DESCRIPTION
**Pull request by albarin**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/46837

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

**If the currently active block theme was already customized, the intro page design should be updated**
1. Go to `wp-admin/tools.php?page=woocommerce-admin-test-helper` click on `Reset Customize Your Store`.
2. Go to `Appearance > Theme` and select the TT4 theme.
3. Go to `WooCommerce > Home`, click on `Customize your store`, then on `Start designing`.
4. Go again to `WooCommerce > Home`, click on `Customize your store`.
5. Make sure the intro page looks like this 👇 
Screenshot 2024-04-30 at 16 33 38

6. Click on `Customize your theme` and check you are redirected to the assembler.
7. Go to `Appearance > Theme` and select a classic theme (like Storefront or TT1).
8. Go again to `WooCommerce > Home`, click on `Customize your store` and `Customize your theme`.
9. Make sure you are now redirected to the Customizer (not the assembler).
10. Go again to `WooCommerce > Home`, click on `Customize your store`.
11. Click on `Browse themes` and make sure you are redirected to `WooCommerce → Extensions → Themes`.
12. Go again to `WooCommerce > Home`, click on `Customize your store`.
13. Click on `Use the store designer` and make sure you are redirected to the Customizer.
14. Go to `Appearance > Theme` and select the TT4 theme.
15. Go again to `WooCommerce > Home`, click on `Customize your store`.
16. Click on `Use the store designer` and make sure you are redirected to the assembler.

**Trying to access the assembler with a different theme than TT4 should ask the user to design a new theme**
1. Go to `wp-admin/tools.php?page=woocommerce-admin-test-helper` and click on `Reset Customize Your Store`.
2. Go to `Appearance > Theme` and select a block theme (other than TT4, like Tsubaki or TT3).
3. Go to `WooCommerce > Home`, click on `Customize your store`, then on `Start designing`.
4. Make sure you see the following modal:
Screenshot 2024-06-05 at 10 02 57

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 
[CYS] Remove the restriction to TT4 and allow users to proceed to the pattern assembler with any block themes. Update intro page design.
#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-admin/client/customize-store/intro/intro-banners.tsx [hunk 1]
```diff
@@ -331,14 +341,21 @@ export const ExistingNoAiThemeBanner = () => {
 			buttonIsLink={ false }
 			bannerButtonOnClick={ () => {
 				trackEvent( 'customize_your_store_intro_customize_click' );
```
Comment (albarin) [inline plugins/woocommerce-admin/client/customize-store/intro/intro-banners.tsx]: @nefeline do you know if we should keep this event or maybe send a different one depending on the type of theme?
Also, do we need any other events for this new design links? TY!

**Review by nefeline — COMMENTED** [review]
Thank you so much for working on this @albarin ! Sharing here my test results:

## With the TT2 theme:

* I was able to access the Assembler normally and change all templates as expected.
* After finishing my customizations and heading to the frontend, I get the following errors:

Screenshot 2024-05-02 at 12 14 21

Screenshot 2024-05-02 at 12 14 30

we might have to open an issue to investigate why these template parts are being deleted.

## When switching from TT2 to TT4 theme

* I noticed that I can't access the assembler anymore after switching my current active theme from TT2 to TT4: the intro screen is just refreshed.

https://github.com/woocommerce/woocommerce/assets/15730971/588c411b-6732-4531-9bb3-c496862c3455

* If I reset the CYS then I'm presented with the initial "Design your own" screen again and I can access the assembler

Considering it is a very common and possible scenario for a user to switch between different themes, I think we should fix this and ensure the Assembler is accessible without the need to reset the CYS, WDYT?

# When using a classic theme

* ✅ Confirmed that I'm redirected to the customizer

## When having a customized block theme

* ✅ Confirmed that I'm redirected to the assembler

## The new cards:

Confirmed that they are both displayed and the links are correct and leading users to the expected path:


Screenshot 2024-05-02 at 12 36 15


Would it be worth adding a margin or padding below these two? Right now they are tied to the bottom of the screen.

File: plugins/woocommerce-admin/client/customize-store/intro/intro-banners.tsx (hunk 1 above)
Comment (nefeline) [reply plugins/woocommerce-admin/client/customize-store/intro/intro-banners.tsx]: > do you know if we should keep this event or maybe send a different one depending on the type of theme?

I don't 🤔 : thinking about this, perhaps we should preserve `customize_your_store_intro_customize_click` as-is and add extra events to check for the theme types. Maybe we could check with Veronica as well? IMO the more information we can collect, the better!

Comment (nefeline) [reply plugins/woocommerce-admin/client/customize-store/intro/intro-banners.tsx]: > Also, do we need any other events for this new design links? TY!

Yes! We should def. add tracking events to the two new cards:

Screenshot 2024-05-02 at 12 36 15

**Comment by albarin** [comment]
> After finishing my customizations and heading to the frontend, I get the following errors:

True! I'm not seeing the errors, but I don't see the header or footers 😱 and it happens with other block themes as well (TT3, Tsubaki, etc).

> I noticed that I can't access the assembler anymore after switching my current active theme from TT2 to TT4: the intro screen is just refreshed.

I could reproduce, thanks.

> Would it be worth adding a margin or padding below these two? Right now they are tied to the bottom of the screen.

Yes, makes sense.

I'll start working into all of these now, thanks for the review 🙏

**Comment by gigitux** [comment]
> True! I'm not seeing the errors, but I don't see the header or footers 😱 and it happens with other block themes as well (TT3, Tsubaki, etc).
> 
> >

There is a dedicated issue for this: https://github.com/woocommerce/woocommerce/issues/42181. Luckly, it looks like that @Aljullu is working on the fix! ❤️

File: plugins/woocommerce-admin/client/customize-store/intro/intro-banners.tsx (hunk 1 above)
Comment (albarin) [reply plugins/woocommerce-admin/client/customize-store/intro/intro-banners.tsx]: I have added events to those 2 new links and added a property with the type of theme to the old `customize_your_store_intro_customize_click`.

https://github.com/woocommerce/woocommerce/pull/46916/commits/2c7be044aa4288100fd5c36eaaab8f37ea5895b5

I'll double check in the channel with Veronica next week when she's back.

**Comment by albarin** [comment]
@nefeline I've changed the check `checkActiveThemeHasMods` for `checkCustomizeStoreCompleted`, so we rely on the same condition everywhere.
https://github.com/woocommerce/woocommerce/pull/46916/commits/ad00be86b591b0b278b5140d15f2c6ddfb4b6721

Also added the margin to the cards.
Could you do another review when you have time? 🙏

**Comment by albarin** [comment]
@nefeline btw, this would get rid of the first deprecation warning (`getCurrentThemeGlobalStylesRevisions`) on this issue https://github.com/woocommerce/woocommerce/issues/47012

**Comment by albarin** [comment]
@gigitux @nefeline I've removed the ability to use any other theme apart from TT4 (context: p1717499442912829-slack-C053716F2H2). I've left most of the code to check the theme there though, since we are going to need it eventually. 
Could you please have another look at the PR? 🙏

**Comment by gigitux** [comment]
Thanks @albarin for working on this! I followed the testing instructions, and everything worked okay except for steps 13 - 15.


In this video, I have Tsubaki as active theme. I'm not sure if this beavhiour is the right one:


https://github.com/woocommerce/woocommerce/assets/4463174/c9faf4a8-beec-4d43-bcc9-a0df243e44ff



If I click on `Customize your theme` and `Use the store designer,` I see that the Assembler is loaded with the Tsubaki default template.

Instead, in this case, I would:

- redirect to Site Editor when the user clicks on `Customize your theme`(it is necessary to update the label too. Currently, it is: `Continue to customize your store using the store designer. Change your color palette, fonts, page layouts, and more.`
- start the CYS setup flow when the user clicks on `Use the store designer`.

**Comment by albarin** [comment]
@gigitux Thanks for the review. Sorry, yesterday I made a change to how this should work but steps 13-15 were not updated accordingly 🙏 I have updated them now, I've also added an extra test section as well to test the assembler cannot be accessed with any block theme which is not TT4.

**Comment by gigitux** [comment]
> @gigitux Thanks for the review. Sorry, yesterday I made a change to how this should work but steps 13-15 were not updated accordingly 🙏 I have updated them now, I've also added an extra test section as well to test the assembler cannot be accessed with any block theme which is not TT4.

No worries! Thanks for updating the testing instructions! In some specific conditions, I'm able to access the Assembler even if I'm not using TT4. If it is not super complex to fix it, I think that we should fix it.

- Start the CYS flow with TT4.
- Save.
- Switch to another block theme.
- Visit `wp-admin/admin.php?page=wc-admin&path=%2Fcustomize-store`
- The Assembler is accessible if I click on `Customize your theme` and `Use the store designer`.


https://github.com/woocommerce/woocommerce/assets/4463174/2b5dbdfa-3826-48a3-ba00-989c0d52f79b

**Comment by albarin** [comment]
@gigitux I've pushed a fixed, TY!

File: plugins/woocommerce/tests/e2e-pw/tests/customize-store/assembler/color-picker.spec.js [hunk 2]
```diff
@@ -300,18 +299,6 @@ test.describe( 'Assembler -> Color Pickers', () => {
...
-
-			const waitResponse = page.waitForResponse(
-				( response ) =>
-					response.url().includes( 'wp-json/wp/v2/global-styles' ) &&
-					response.status() === 200
-			);
-
-			await saveButton.click();
-
-			await waitResponse;
```
Comment (gigitux) [inline plugins/woocommerce/tests/e2e-pw/tests/customize-store/assembler/color-picker.spec.js:313]: Is there a particular reason for removing this?

File: plugins/woocommerce/tests/e2e-pw/tests/customize-store/assembler/color-picker.spec.js [hunk 3]
```diff
@@ -300,18 +299,6 @@ test.describe( 'Assembler -> Color Pickers', () => {
...
-
```
Comment (gigitux) [inline plugins/woocommerce/tests/e2e-pw/tests/customize-store/assembler/color-picker.spec.js]: Is there a particular reason for removing this?

**Review by gigitux — APPROVED** [review]
Thanks for working on this! Everything LGTM now! Before merging this PR, can you:
- restore the changes that you did to the `color-picker` E2E test? It looks they are unrelated to this PR 🤔 
- fix lint errors

🚀 

Thanks!

**Review by nefeline — APPROVED** [review]
Thanks for your amazing work here, @albarin ! I've ran all tests and confirmed everything works as described. I just have a note regarding the following step with classic themes:

> Click on Use the store designer and make sure you are redirected to the Customizer.

Not at all related to your work here, but it feels a bit odd to have two links on the same page directing the user to the customizer. This is more of a UX concern & not related to the implementation, but probably worth considering a discussion whenever we start working with extensibility.

File: plugins/woocommerce/tests/e2e-pw/tests/customize-store/assembler/color-picker.spec.js (hunk 3 above)
Comment (albarin) [reply plugins/woocommerce/tests/e2e-pw/tests/customize-store/assembler/color-picker.spec.js]: No, not sure what happened, I've restored it and fixed the lint errors.

File: plugins/woocommerce/tests/e2e-pw/tests/customize-store/assembler/color-picker.spec.js [hunk 4]
```diff
@@ -287,9 +287,9 @@ test.describe( 'Assembler -> Color Pickers', () => {
 	for ( const [ colorPaletteName, colors ] of Object.entries(
 		colorPalette
 	) ) {
-		test( `Color palette ${ colorPaletteName } should be applied`, async ( {
+		test.only( `Color palette ${ colorPaletteName } should be applied`, async ( {
```
Comment (gigitux) [inline plugins/woocommerce/tests/e2e-pw/tests/customize-store/assembler/color-picker.spec.js]: I think that we should remove `only` too:
```suggestion
		test( `Color palette ${ colorPaletteName } should be applied`, async ( {
```

**Comment by jsnajdr** [comment]
Hi @albarin :wave: The `professional-theme.svg` image that this PR is adding is super big, 24MB. It's an SVG that embeds four PNG images, 4076x2426px, 6MB each:

Screenshot 2024-06-18 at 11 46 31

Let's optimize it. After all, it ends up being used as a 170x100px illustration for the "Choose a professionally designed theme" link.

**Comment by albarin** [comment]
Sorry @jsnajdr, totally missed this. I've just opened a PR to optimize the image: https://github.com/woocommerce/woocommerce/pull/48566

**Pull request merged**